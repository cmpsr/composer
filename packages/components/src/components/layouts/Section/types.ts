import { BoxProps } from '@chakra-ui/layout';

export interface SectionProps extends BoxProps {}
