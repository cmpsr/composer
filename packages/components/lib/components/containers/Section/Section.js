"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Section = exports.SECTION_DEFAULT_TEST_ID = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const SECTION_DEFAULT_TEST_ID = 'section';
exports.SECTION_DEFAULT_TEST_ID = SECTION_DEFAULT_TEST_ID;

const Section = ({
  children,
  className,
  testId = SECTION_DEFAULT_TEST_ID
}) => {
  return /*#__PURE__*/_react.default.createElement("section", {
    className: (0, _classnames.default)('mb-4', className),
    "data-testid": testId
  }, children);
};

exports.Section = Section;