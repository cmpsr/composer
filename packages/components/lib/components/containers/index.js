"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Box = require("./Box");

Object.keys(_Box).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Box[key];
    }
  });
});

var _Grid = require("./Grid");

Object.keys(_Grid).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Grid[key];
    }
  });
});

var _Section = require("./Section");

Object.keys(_Section).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Section[key];
    }
  });
});

var _SectionChild = require("./SectionChild");

Object.keys(_SectionChild).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SectionChild[key];
    }
  });
});