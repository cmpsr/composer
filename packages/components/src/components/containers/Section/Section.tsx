import React, { ReactNode } from 'react';
import cn from 'classnames';

export const SECTION_DEFAULT_TEST_ID = 'section';

type Props = {
  children?: ReactNode;
  className?: string;
  testId?: string;
};

export const Section = ({
  children,
  className,
  testId = SECTION_DEFAULT_TEST_ID,
}: Props) => {
  return (
    <section className={cn('mb-4', className)} data-testid={testId}>
      {children}
    </section>
  );
};
