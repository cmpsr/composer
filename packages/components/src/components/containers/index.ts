export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
