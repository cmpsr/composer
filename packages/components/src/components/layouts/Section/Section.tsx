import React, { ReactElement } from "react";
import cn from "classnames";

import {
  Section as SectionContainer,
  Box,
  SectionChild,
} from "components/containers";
import { SectionStyle } from "types";

type Props = {
  children?: {
    title?: ReactElement;
    subtitle?: ReactElement;
    content?: ReactElement | ReactElement[];
  };
  className?: string;
  style?: SectionStyle;
};

export const Section = ({ children, className }: Props) => {
  return (
    <SectionContainer className={cn(className, "px-4 md:px-8")}>
      {children.title && <Box size="content">{children.title}</Box>}
      {children.subtitle && <Box size="content">{children.subtitle}</Box>}
      {children.content &&
        React.Children.map(children.content, (child) => (
          <SectionChild>{child}</SectionChild>
        ))}
    </SectionContainer>
  );
};
