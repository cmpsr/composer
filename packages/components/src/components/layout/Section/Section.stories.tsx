import React from 'react';
import { Meta } from '@storybook/react';
import { Section, SectionProps } from './';
import { Text } from 'components';

export default {
  title: 'Components/Layout/Section',
  component: Section,
} as Meta;

interface StoryProps extends SectionProps {
  content: string;
}

const Template: React.FC<StoryProps> = ({ content, ...rest }) => (
  <Section {...rest}>
    <Text variant="font-body-display-L">{content}</Text>
  </Section>
);
export const Playground = Template.bind({});
Playground.args = {
  padding: '1rem',
  border: '1px solid',
  borderColor: 'primary-ui-default',
  content: 'Composer rocks!',
  width: '100%',
};
