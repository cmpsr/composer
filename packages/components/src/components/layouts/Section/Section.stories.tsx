import React from 'react';
import { Meta } from '@storybook/react';
import { Section, SectionProps } from '.';
import { Text } from 'components';

export default {
  title: 'Components/Layouts/Section',
  component: Section,
} as Meta;

interface StoryProps extends SectionProps {
  content: string;
}

const Template: React.FC<StoryProps> = ({ content, ...rest }) => (
  <Section {...rest}>
    <Text variant="text-body-display-L">{content}</Text>
  </Section>
);
export const Playground = Template.bind({});
Playground.args = {
  padding: 'spacer-4',
  border: '1px solid',
  borderColor: 'accent-default',
  content: 'Composer rocks!',
  width: '100%',
};
