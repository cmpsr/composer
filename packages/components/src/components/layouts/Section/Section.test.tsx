import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Section } from './Section';
import { SECTION_DEFAULT_TEST_ID } from '../../containers/Section';

describe('Section', () => {
  const children = {
    title: <div>test title</div>,
    subtitle: <div>test subtitle</div>,
    content: <div>test-content</div>,
  };

  it('should match snapshot', () => {
    const section = renderer.create(
      <Section className="foo" children={children} />,
    );
    const tree = section.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should render section component', () => {
    const { getByTestId } = render(
      <Section className="foo" children={children} />,
    );
    const section = getByTestId(SECTION_DEFAULT_TEST_ID);
    expect(section.classList).toContain('foo');
  });
  it('should render component with test title', () => {
    const children = {
      title: <div>test title</div>,
    };

    const { getByText } = render(
      <Section className="foo" children={children} />,
    );
    const box = getByText('test title');
    expect(box).toBeInTheDocument();
  });
  it('should render component with subtitle', () => {
    const children = {
      subtitle: <div>subtitle test</div>,
    };

    const { getByText } = render(
      <Section className="foo" children={children} />,
    );
    const box = getByText('subtitle test');
    expect(box).toBeInTheDocument();
  });
  it('should render component with content', () => {
    const children = {
      content: <div>content test</div>,
    };

    const { getByText } = render(
      <Section className="foo" children={children} />,
    );
    const box = getByText('content test');
    expect(box).toBeInTheDocument();
  });
});
