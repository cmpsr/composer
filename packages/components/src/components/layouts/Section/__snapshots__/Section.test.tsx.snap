// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should match snapshot 1`] = `
<section
  className="mb-4 foo px-4 md:px-8"
  data-testid="section"
>
  <div
    className="w-full h-full max-w-content mx-auto"
    data-testid="box"
  >
    <div>
      test title
    </div>
  </div>
  <div
    className="w-full h-full max-w-content mx-auto"
    data-testid="box"
  >
    <div>
      test subtitle
    </div>
  </div>
  <div
    className="py-8"
  >
    <div>
      test-content
    </div>
  </div>
</section>
`;
